File: Data Structures & Algorithms/depth-of-binary-tree/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxDepth(TreeNode* root) {
        queue<TreeNode*>q;
        int level=0;
        if(root!=NULL)
        {
            q.push(root);
        }
        while(q.empty()==0)
        {
            int size=q.size();
            for(int i=0;i<size;i++)
            {
                TreeNode* front=q.front();
                q.pop();
                if(front->left)
                {
                    q.push(front->left);
                }
                if(front->right)
                {
                    q.push(front->right);
                }
            }
            level++;
            
        }
        return level;
        
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        
        Queue<TreeNode>q=new ArrayDeque<>();
        List<List<Integer>>mainlist=new ArrayList<>();
        if(root==null)
        {
            return mainlist;
        }
        q.offer(root);
        while(!q.isEmpty())
        {
            int size=q.size();
            List<Integer>l=new ArrayList<>();
            for(int i=0;i<size;i++)
            {
                TreeNode curr=q.poll();
                l.add(curr.val);
                if(curr.left!=null)
                {
                    q.offer(curr.left);
                }
                if(curr.right!=null)
                {
                    q.offer(curr.right);
                }
            }
            mainlist.add(l);
        }
        return mainlist;
        
    }
}
